### DISCUSSION TITLE
Sonoff 4chproR3

### DISCUSSION THREAD

**Discussion by Stephan56370** [arendst/Tasmota#12096] [Q&A] [answered]
Hello,
I'm trying to have one of the button of the sonoff to act as a pushbutton (relay on when button is pressed, relay off when button is released)
In fact, it almost work like that out of the box.
But it seems that that are 2 behaviors depending on how long time you press the button.
If I make a long press then it work as discribed and as I want.
If I make a short press (say less than a sec) then the relay change its state (toggle mode) and this is what I would like to avoid

How can I change the behavior when I short press the button ?
or where can I set the time to 0, so the behavior could be the same wether I long or short press the button ?

**Comment by ascillato** [comment, marked as answer]
Please, check the docs at `switchmode` command.

**Reply by Stephan56370** [reply]
Thank you for trying to help

I checked and tried all modes with no success, anyway it seems to me that switchmodes does not describe nor make any difference between short and long press.
I also tried to add rules like On button1#state=0 do power1 0 endon to force the relay to go off when button is relased.

-may be, I do not know how to go from a button to switch definition ? and the button on sonoff is defined as a button and not as a switch ?
- may be the rules are not actives, I do not know how to check wether it is active or not.

**Reply by ascillato** [reply]
> relay on when button is pressed, relay off when button is released

Disable your rules using command `backlog Rule1 0;Rule2 0;Rule3 0` in the console
Delete any Switchtopic you could have using command `switchtopic 0` in the console

Please, in the Tasmota Web UI:

1- go to CONFIGURATION then CONFIGURE TEMPLATE.
2- on the dropdown called BASED ON, select SONOFF CH PRO (23)
3- on the GPIO 0, change BUTTON 1 to SWITCH 1
4- click SAVE
5- Go to CONFIGURATION then CONFIGURE OTHER
6- Click on ACTIVATE on the TEMPLATE BOX
7- Click SAVE
8- Go to console and type `SWITCHMODE1 1`

**Reply by Stephan56370** [reply]
Perfect, 
I thought I already tryed that but I followed anyway your nice step by step guide.
I believe, resetting and starting from scratch was the key

Thank you very very much, it now works !
